src/java.base/share/classes/java/util/zip/Deflater.java

```diff
@@ -60,8 +60,10 @@
  * {@snippet id="compdecomp" lang="java" class="Snippets" region="DeflaterInflaterExample"}
  *
  * @apiNote
- * To release resources used by this {@code Deflater}, the {@link #end()} method
- * should be called explicitly. Subclasses are responsible for the cleanup of resources
+ * The {@code Deflater} cleans up its resources in its {@link #end()} method. To facilitate
+ * usage with try-with-resources, this class implements {@link AutoCloseable}. The
+ * {@link #close()} method of this class calls {@code end()} to clean up its
+ * resources. Subclasses are responsible for the cleanup of resources
```

Comment:
As part of the clarification, do we need to state the verbiage above regarding try-with-resources specifically?

I did a quick scan of some of the other classes implementing AutoClosable and did not see many cases of that in the documentation most likely due to that behavior is clear in the documentation for AutoClosable

Reply:
Stuart in one of his review comments in this PR had noted that:

> I think the class specification needs to be clearer about the positioning of end() and close(). The end() method has done the real work of "closing" since the classes were introduced. We're retrofitting them to to have a close() method that's essentially just a synonym for end(), and it's still perfectly legal for clients and subclasses to call end() instead of close(). I think we need to say that close() is present mainly to support AutoCloseable and try-with-resources.

The current wording of that class level javadoc is my attempt to implement that suggestion. Do you think we can word it differently and yet convey the purpose of close()?

Reply:
Hi Jai,

I am not convinced we need to call this out specifically at the class level given  the classes implements AutoClosable and close indicates it calls end.  We will also  want to add a Release Note which can help further clarify this

Now that being said, what I would suggest instead is a simple code example in the class header which using try-with-resources would be more effective/helpful.

Reply:
Hello Stuart, I've now updated the PR to follow your recommendations for the class level javadoc of the `Deflater`. Does this look OK?

Like you note, I'll update the `Inflater` similarly once we settle on the text for the `Deflater`.